symengine/derivative.cpp

```diff
@@ -80,6 +80,25 @@ class DiffImplementation
                    self.get_a()->diff(x));
     }
 
+    static RCP<const Basic> diff(const Dirichlet_eta &self,
+                                 const RCP<const Symbol> &x)
+    {
+        RCP<const Basic> m1;
+        m1 = zeta(self.get_arg())->diff(x);
+        // check https://en.wikipedia.org/wiki/Dirichlet_eta_function
+        // for the equation
+        if (eq(*self.get_arg(), *one)) {
```

Comment:
as zeta(1)-->infinity and zeta'(1)--> -infinity the derivative is not defined at 1. So this is limit is calculated using Laurent expantion as you can read here:
http://math.stackexchange.com/questions/253549/problem-with-the-dirichlet-eta-function

or maybe I can remove this condition?